iceoryx_hoofs/test/moduletests/test_cxx_function_ref.cpp

```diff
@@ -210,20 +163,16 @@ TEST_F(function_refTest, CreateValidByMoveAssignResultEqual)
         function_ref<int()> sut2{lambda2};
         sut1 = std::move(sut2);
     }
-    EXPECT_TRUE(sut1);
     EXPECT_THAT(sut1(), Eq(999));
 }
 
-TEST_F(function_refTest, CreateInvalidByMoveAssignIsFalse)
+TEST_F(function_refDeathTest, CallMovedFromLeadsToTermination)
 {
-    ::testing::Test::RecordProperty("TEST_ID", "7dabac7a-dc3d-415f-8250-363ee196ebb8");
-    auto lambda1 = [] {};
-    function_ref<void()> sut1{lambda1};
-    {
-        function_ref<void()> sut2;
-        sut1 = std::move(sut2);
-    }
-    EXPECT_FALSE(sut1);
+    ::testing::Test::RecordProperty("TEST_ID", "3402f27e-ced5-483f-ab39-0069cfd172ac");
+    auto lambda = []() -> int { return 7654; };
+    function_ref<int()> sut1{lambda};
+    function_ref<int()> sut2{std::move(sut1)};
```

Comment:
Is this death accomplished by explicit check for invalid (e.g. nullptr) or by setting a pointer to null and crashing while dereferencing if accessed after move? I would prefer the latter for efficiency. This is legit and STL style (no performance hit when used correctly, guaranteed crash on all platforms if not).

Reply:
Yes, correct the latter.